src/Grid/grid_generators.jl

```diff
@@ -0,0 +1,386 @@
+export generate_grid
+
+# Goes through `boundary` and updates the onboundary values for the
+# cells that actually are on the boundary
+function _apply_onboundary!{T}(::Type{T}, cells, boundary)
+    cells_new = T[]
+    isboundary = zeros(Bool, nfaces(T))
+    for faceindex in boundary
+        fill!(isboundary, false)
+        cell, face = faceindex
+        c = cells[cell]
+        for i in 1:nfaces(T)
+            isboundary[i] = onboundary(c, i) | (face == i)
+        end
+        cells[cell] = T(c.nodes, isboundary)
+    end
+end
+
+"""
+`Grid` generator for a rectangle in 1, 2 and 3 dimensions.
+
+    generate_grid(celltype::Cell{dim, N}, nel::NTuple{dim, Int}, [left::Vec{1, T}=Vec{1}((-1.0,)), right::Vec{1, T}=Vec{1}((1.0,))])
+
+**Arguments**
+
+* `celltype`: a celltype, e.g. `Triangle` or `Hexahedron`
+* `nel`: a tuple with number of elements in each direction.
+* `left`, `right`: optional endpoints of the domain, defaults to `-one(Vec{dim})` and `one(Vec{dim})`
+
+**Results**
+
+* `grid`: a `Grid`.
+
+"""
+# Line
+function generate_grid{T}(::Type{Line}, nel::NTuple{1, Int}, left::Vec{1, T}=Vec{1}((-1.0,)), right::Vec{1, T}=Vec{1}((1.0,)))
+    nel_x = nel[1]
+    n_nodes = nel_x + 1
+
+    # Generate nodes
+    coords_x = collect(linspace(left[1], right[1], n_nodes))
+    nodes = Node{1,T}[]
+    for i in 1:n_nodes
+        push!(nodes, Node((coords_x[i],)))
+    end
+
+    # Generate cells
+
+    cells = Line[]
+    for i in 1:nel_x
+        push!(cells, Line((i, i+1)))
+    end
+
+
+    # Cell faces
+    boundary = Vector([(1, 1),
+                       (nel_x, 2)])
+
+    _apply_onboundary!(Line, cells, boundary)
+
+    # Cell face sets
+    facesets = Dict("left"  => Set{Tuple{Int, Int}}([boundary[1]]),
+                    "right" => Set{Tuple{Int, Int}}([boundary[2]]))
+    return Grid(cells, nodes, facesets=facesets)
+end
+
+# QuadraticLine
+function generate_grid{T}(::Type{QuadraticLine}, nel::NTuple{1, Int}, left::Vec{1, T}=Vec{1}((-1.0,)), right::Vec{1, T}=Vec{1}((1.0,)))
+    nel_x = nel[1]
+    n_nodes = 2*nel_x + 1
+
+    # Generate nodes
+    coords_x = collect(linspace(left[1], right[1], n_nodes))
+    nodes = Node{1,T}[]
+    for i in 1:n_nodes
+        push!(nodes, Node((coords_x[i],)))
+    end
+
+    # Generate cells
+    cells = QuadraticLine[]
+    for i in 1:nel_x
+        push!(cells, QuadraticLine((2*i-1, 2*i+1, 2*i)))
+    end
+
+    # Cell faces
+    boundary = Tuple{Int, Int}[(1, 1),
+                         (nel_x, 2)]
+
+    _apply_onboundary!(QuadraticLine, cells, boundary)
+
+    # Cell face sets
+    facesets = Dict("left"  => Set{Tuple{Int, Int}}([boundary[1]]),
+                    "right" => Set{Tuple{Int, Int}}([boundary[2]]))
+    return Grid(cells, nodes, facesets=facesets)
+end
+
+function _generate_2d_nodes!(nodes, nx, ny, LL, LR, UR, UL)
```

Comment:
It is not possible to give all four corners in the 2d grid generators (as in the COok membrane notebook). Should probably extend to 3d as well.